uberpoet/cpprojectgen.py

```diff
@@ -186,10 +168,11 @@ def gen_app_main(self, app_node, module_index):
 
     # Library Generation
 
-    def gen_lib_module(self, module_node, loc_per_unit, language):
+    def gen_lib_module(self, module_index, module_node, loc_per_unit, language):
         # Make Podspec Text
         deps = self.make_dep_list([i.name for i in module_node.deps])
         pod_text = self.pod_lib_template.format(module_node.name, deps, self.wmo_state)
+        deps_from_index = [{n.name: module_index[n.name]} for n in module_node.deps]
```

Comment:
since we now return a topologically sorted list of the graph it means that we will already have the deps of a module inside the module index before we process it. This allows us to reach into the generated sources for the dependencies in order to create an instance of their class and invoke their functions.

Reply:
I will add this as a comment too.